zavod/zavod/sentry.py

```diff
@@ -0,0 +1,272 @@
+        if ignore_loggers is not None:
+            self._ignored_loggers.update(set(ignore_loggers))
+
+    @staticmethod
+    def _get_logger_name(
+        logger: WrappedLogger, event_dict: MutableMapping[str, Any]
+    ) -> Optional[str]:
+        """Get logger name from event_dict with a fallbacks to logger.name and
+        record.name
+
+        :param logger: logger instance
+        :param event_dict: structlog event_dict
+        """
+        record = event_dict.get("_record")
+        l_name = event_dict.get("logger")
+        logger_name = None
+
+        if l_name:
+            logger_name = l_name
+        elif record and hasattr(record, "name"):
+            logger_name = record.name
+
+        if not logger_name and logger and hasattr(logger, "name"):
+            logger_name = logger.name
+
+        return logger_name
+
+    def _get_scope(self) -> Scope:
+        return self._scope or get_isolation_scope()
+
+    def _get_event_and_hint(
+        self, event_dict: EventDict
+    ) -> tuple[Event, dict[str, Any]]:
+        """Create a sentry event and hint from structlog `event_dict` and sys.exc_info.
+
+        :param event_dict: structlog event_dict
+        """
+        exc_info = _figure_out_exc_info(event_dict.get("exc_info", None))
+        has_exc_info = exc_info and exc_info != (None, None, None)
+
+        if has_exc_info:
+            client = self._get_scope().get_client()
+            options: dict[str, Any] = client.options if client else {}
+            event, hint = event_from_exception(
+                exc_info,
+                client_options=options,
+            )
+        else:
+            event, hint = {}, {}
+
+        event["message"] = str(event_dict.get("event"))
+        event["level"] = event_dict.get("level")  # type: ignore
+        if self._fingerprint is not None:
+            event["fingerprint"] = self._fingerprint
+        # We push the dataset into the logger field because that shows up in the issues overview list
+        # and there doesn't seem to be a good way to change that.
+        if "dataset" in event_dict:
+            event["logger"] = event_dict["dataset"]
+
+        if self._event_dict_as_extra:
+            # Only add the keys we're not already reporting in other fields to extra data
+            event["extra"] = {
+                k: v
+                for k, v in event_dict.items()
+                if k not in ["event", "level", "logger"]
+            }
+
+        if self.tag_keys == "__all__":
+            event["tags"] = self._original_event_dict.copy()
+        if isinstance(self.tag_keys, list):
+            event["tags"] = {
+                key: event_dict[key] for key in self.tag_keys if key in event_dict
+            }
+        # Because we push the dataset into the Event.logger field (see above), we push the logger field into the tags.
+        if "logger" in event_dict:
+            event["tags"]["python_logger"] = event_dict["logger"]
```

Comment:
sorry for the late one - this feels like a bit of a hack. We might also miss out on built-in sentry functionality if we hack the logger name.

Is there a less hacky way to just make sure dataset name is in the message like `[dataset_name] Something happened` _and_ tags?

Reply:
You're completely right, fixed as suggested.